react/src/tests/opReturn.test.ts

```diff
@@ -85,94 +107,109 @@ describe('getDataPushdata', () => {
+  });
   it('Simple parse OpReturn 1', () => {
-    const fullResult = parseOpReturnProps('myCustomUserData') // 16 bytes
-    const resultPaymentId = fullResult?.slice(-paymentIdDigits)
-    const resultNoPaymentId = fullResult?.slice(0,-paymentIdDigits)
-    expect(resultNoPaymentId).toBe(allResultsPrefix +
-      '10' + // 16 in hex
-      '6d79437573746f6d5573657244617461')
-    expect(resultPaymentId?.length).toBe(paymentIdDigits)
-    expect(resultPaymentId).toMatch(paymentIdRegex)
-  })
+    const fullResult = parseOpReturnProps('myCustomUserData'); // 16 bytes
+    const resultPaymentId = fullResult?.slice(-paymentIdDigits);
+    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits);
+    expect(resultNoPaymentId).toBe(
+      allResultsPrefix +
+        '10' + // 16 in hex
+        '6d79437573746f6d5573657244617461',
+    );
+    expect(resultPaymentId?.length).toBe(paymentIdDigits);
+    expect(resultPaymentId).toMatch(paymentIdRegex);
+  });
   it('Simple parse OpReturn 1 no paymentId', () => {
-    const fullResult = parseOpReturnProps('myCustomUserData', true) // 16 bytes
-    expect(fullResult).toBe(allResultsPrefix +
-      '10' + // 16 in hex
-      '6d79437573746f6d5573657244617461')
-  })
+    const fullResult = parseOpReturnProps('myCustomUserData', true); // 16 bytes
+    expect(fullResult).toBe(
+      allResultsPrefix +
+        '10' + // 16 in hex
+        '6d79437573746f6d5573657244617461' +
+        '00', // Empty nonce pushdata
+    );
+  });
   it('Simple parse OpReturn 2', () => {
-    const fullResult = parseOpReturnProps('my=Longer more=sensible|user|data') // 33 bytes
-    const resultPaymentId = fullResult?.slice(-paymentIdDigits)
-    const resultNoPaymentId = fullResult?.slice(0,-paymentIdDigits)
-    expect(resultNoPaymentId).toBe(allResultsPrefix +
-      '21' + // 33 in hex
-      '6d793d4c6f6e676572206d6f72653d73656e7369626c657c757365727c64617461')
-    expect(resultPaymentId?.length).toBe(paymentIdDigits)
-    expect(resultPaymentId).toMatch(paymentIdRegex)
-  })
+    const fullResult = parseOpReturnProps('my=Longer more=sensible|user|data'); // 33 bytes
+    const resultPaymentId = fullResult?.slice(-paymentIdDigits);
+    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits);
+    expect(resultNoPaymentId).toBe(
+      allResultsPrefix +
+        '21' + // 33 in hex
+        '6d793d4c6f6e676572206d6f72653d73656e7369626c657c757365727c64617461',
+    );
+    expect(resultPaymentId?.length).toBe(paymentIdDigits);
+    expect(resultPaymentId).toMatch(paymentIdRegex);
+  });
   it('Simple parse OpReturn 2 no paymentId', () => {
-    const fullResult = parseOpReturnProps('my=Longer more=sensible|user|data', true) // 33 bytes
-    expect(fullResult).toBe(allResultsPrefix +
-      '21' + // 33 in hex
-      '6d793d4c6f6e676572206d6f72653d73656e7369626c657c757365727c64617461')
-  })
+    const fullResult = parseOpReturnProps(
+      'my=Longer more=sensible|user|data',
```

Comment:
Is this supposed to represent a JSON? Does PayButton have an internal way of consistently encoding / decoding this stuff?

Reply:
Yeah this will be on the docs page once that's updated.

If we can decode it with our scheme here, or if they just enter straight JSON (server PR is coming for that one) then it'll save it as JSON. If it can't decode it then it'll just go in as a string.

Reply:
ok. Ideally we should have a standardized way to encode/decode this data in the spec. utf8 is probably not the best way to do this but it will work.

However, this data can be anything (for now) per spec. It's up to PayButton to optimize how the space is used.

Probably do not want to use space characters (i.e. what is now separating `my=Long` from `more=sensible...`) as the standard delimiter since I can imagine spaces being common in string values. What if the user wants my to actually `my="Longer word than just Longer`? Will this be handled by the current parsing?

Don't need a complicated scheme for encoding and decoding this data but this will also need a unit tested function with all of the cases you support. Outside scope of this diff but just as important to ensuring the txs are useful once they start firing.